lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -432,33 +432,75 @@ def operations
       end
 
       define_method("find_all_of_type_#{pe_type}") do |options = {}|
```

Comment:
This 70-line method is complex and difficult to review. I just checked, and nowhere in dalton do we use `find_all(..., ignore_case: true)`. We also don't appear to use `conditions` or `included_conditions`, if I'm understanding them correctly. It looks like we may have overdesigned this method to begin with, and now we're trying to maintain backwards compatibility for a bunch of functionality no one is using.

I'd be in favor of bumping the pm a major version and making the backwards incompatible change of simplifying this method to:

```ruby
define_method("find_all_of_type_#{pe_type}") do |options = {}|
  options[:unique_identifier] = options.delete(:uuid) if options[:uuid]
  class_for_type(pe_type).where(options.slice(*PolicyElement.column_names))
end
```

We _would_ have to maintain backwards compatibility for `extra_attributes`, so we'd need to tack that in somewhere as well, but I'd focus on simplifying that piece (instead of making it performant) because that bit already doesn't scale in the current implementation.

Reply:
`conditions` is essentially the hash that is used to filter column attributes. The main lookup of this method is in the first `all` variable assignment; the assignment maps over the hash and reduces it to an ActiveRecord_Relation object. 

Unfortunately it appears that we do use `included_conditions`. At first glance it seems the permissions controller uses it; I would need to check thoroughly to find other cases if they exist. 

Additionally it seems that `ignore_case` is set to `[:name]` by default in APE::Base. I haven't seen any other use for it except when `ignore_case` is set to false. I believe the reason for most of these explicit false settings were due to a previous bug in this method that sometimes didn't return results at all unless `ignore_case` was false. 

I definitely agree that this method could be trimmed. I support a major version bump to go with that cleanup, and removing a lot of the unused functionality. That being said, I think there are immediate benefits to be had with even these changes, namely being able to pass in arrays of UUIDs so we don't have to hit the db multiple times and map over UUIDs if we're doing mass lookups, like in CRS#create and #update. 

In my opinion, we should integrate these changes so we can start using the improved functionality with a minor version bump, then create a story to 1. simplify this method considerably and 2. clean up the usage in Dalton to take advantage of the improved method.

Reply:
@DMcKinnon-mdsol I've extracted the logic into several methods

Reply:
`ignore_case: true` is default, so Dalton uses it almost _everywhere_

Reply:
What do you mean @fpcyan ? From what I've seen, the default value of `ignore_case` that is passed into this method is `[:name]`, but I might have missed a default somewhere else.